crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -1688,6 +1738,24 @@ impl<'a> Checker<'a> {
         }
     }
 
+    fn visit_deferred_type_param_definitions(&mut self) {
+        while !self.deferred.type_param_definitions.is_empty() {
+            let type_params = std::mem::take(&mut self.deferred.type_param_definitions);
+            for (type_param, snapshot) in type_params {
+                self.semantic.restore(snapshot);
+
+                match type_param {
+                    ast::TypeParam::TypeVar(ast::TypeParamTypeVar { bound, .. }) => {
+                        if let Some(bound) = bound {
```

Comment:
Nit: you might be able to do this in the match? Like `ast::TypeParam::TypeVar(ast::TypeParamTypeVar { Some(bound), .. }) => {`? But I didn't try it.

Reply:
I think it's boxed so it doesn't work.

Reply:
Not related to the box but it does fail
```❯ cc
    Checking ruff v0.0.280 (/Users/mz/eng/src/astral-sh/ruff/crates/ruff)
error: expected `,`
    --> crates/ruff/src/checkers/ast/mod.rs:1748:73
     |
1748 |                     ast::TypeParam::TypeVar(ast::TypeParamTypeVar { Some(bound), .. }) => {
     |                                             ---------------------       ^
     |                                             |
     |                                             while parsing the fields for this pattern

error[E0425]: cannot find value `bound` in this scope```

Reply:
Ah duh

```rust
ast::TypeParam::TypeVar(ast::TypeParamTypeVar {
                        bound: Some(bound), ..
                    }) => {
```